File: Bwl.Network.ClientServer.Repeater/RepeaterInterface.vb
Imports System.Timers

Public Class RepeaterInterface
    Private WithEvents _stateTimer As New Timer(3000)
    Private _logger As Logger
    Private _port As IntegerSetting
    Private _formDescriptor As AutoFormDescriptor
    Private _clientsList As AutoListbox
    Private _showDumps As Boolean
    Private WithEvents _showDumpsButtons As AutoButton
    Private _server As NetServer
    Private _core As RepeaterCore

    Public Sub New(app As AppBase, core As RepeaterCore)
        _core = core
        _server = core.NetServer
        _formDescriptor = New AutoFormDescriptor(app.AutoUI, "Repeater Form") With {.FormHeight = 500, .LoggerExtended = False}
        _clientsList = New AutoListbox(app.AutoUI, "Connected Clients")
        _showDumpsButtons = New AutoButton(app.AutoUI, "Show\Hide Dumps")
        _clientsList.Info.Width = 400
        _logger = app.RootLogger
        _logger.AddMessage("Created autointerface")
        _stateTimer.AutoReset = True
        _stateTimer.Start()
    End Sub

    Private Sub _stateTimer_Elapsed(sender As Object, e As ElapsedEventArgs) Handles _stateTimer.Elapsed
        Try
            Dim clients = _server.Clients.ToArray
            Dim items As New List(Of String)
            For Each client In clients
                items.Add("#" + client.ID.ToString + ", [" + client.RegisteredID + "], " + client.IPAddress + ", Received\Sent: " + client.ReceivedMessages.ToString + "\" + client.SentMessages.ToString)
            Next
            _clientsList.Items.Replace(items.ToArray)
        Catch ex As Exception
        End Try
    End Sub

    Private Sub _showDumpsButtons_Click(source As AutoButton) Handles _showDumpsButtons.Click
        _core.LogMessages = Not _core.LogMessages
    End Sub
End Class

File: Bwl.Network.ClientServer.Repeater.Avalonia/RepeaterInterface.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Runtime.CompilerServices;
using System.Timers;

namespace Bwl.Network.ClientServer
{

    public class RepeaterInterface
    {
        private Timer _stateTimer;
        private Bwl.Framework.Logger _logger;
        private Bwl.Framework.IntegerSetting _port;
        private Bwl.Framework.AutoFormDescriptor _formDescriptor;
        private Bwl.Framework.AutoListbox _clientsList;
        private bool _showDumps;
        private Bwl.Framework.AutoButton __showDumpsButtons;

        private Bwl.Framework.AutoButton _showDumpsButtons
        {
            [MethodImpl(MethodImplOptions.Synchronized)]
            get
            {
                return __showDumpsButtons;
            }

            [MethodImpl(MethodImplOptions.Synchronized)]
            set
            {
                if (__showDumpsButtons != null)
                {
                    __showDumpsButtons.Click -= _showDumpsButtons_Click;
                }

                __showDumpsButtons = value;
                if (__showDumpsButtons != null)
                {
                    __showDumpsButtons.Click += _showDumpsButtons_Click;
                }
            }
        }
        private ClientServer.NetServer _server;
        private ClientServer.RepeaterCore _core;

        public RepeaterInterface(Bwl.Framework.AppBase app, ClientServer.RepeaterCore core)
        {
            _stateTimer = new Timer(3000d);
            _core = core;
            _server = core.NetServer;
            _formDescriptor = new Bwl.Framework.AutoFormDescriptor(app.AutoUI, "Repeater Form") { FormHeight = 500, LoggerExtended = false };
            _clientsList = new Bwl.Framework.AutoListbox(app.AutoUI, "Connected Clients");
            _showDumpsButtons = new Bwl.Framework.AutoButton(app.AutoUI, @"Show\Hide Dumps");
            _clientsList.Info.Width = 500;
            _logger = app.RootLogger;
            _logger.AddMessage("Created autointerface");
            _stateTimer.AutoReset = true;
            _stateTimer.Start();
            _stateTimer.Elapsed += _stateTimer_Elapsed;
        }

        private void _stateTimer_Elapsed(object sender, ElapsedEventArgs e)
        {
            try
            {
                ClientServer.ConnectedClient[] clients = _server.Clients.ToArray();
                var items = new List<string>();
                foreach (var client in clients)
                    items.Add("#" + client.ID.ToString() + ", [" + client.RegisteredID + " @ " + client.RegisteredServiceName + "], " + client.IPAddress + @", Received\Sent: " + client.ReceivedMessages.ToString() + @"\" + client.SentMessages.ToString());
                _clientsList.Items.Replace(items.ToArray());
            }
            catch (Exception ex)
            {
            }
        }

        private void _showDumpsButtons_Click(Bwl.Framework.AutoButton source)
        {
            _core.LogMessages = !_core.LogMessages;
        }
    }
}